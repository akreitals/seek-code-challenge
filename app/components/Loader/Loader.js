import React from 'react';
import styled, { keyframes } from 'styled-components';
import Box from '../Box/Box';
import { colors } from '../theme';

const LoaderBounce = keyframes`
    33% {
        -webkit-transform: translateY(-18px);
    }
    66% {
        -webkit-transform: translateY(0);
    }
`;

const LoaderBall = styled('div')`
    background-color: ${colors.midGrayLight};
    width: 18px;
    height: 18px;
    margin: 0 2px;
    border-radius: 100%;
    display: inline-block;

    &:first-child {
        animation: ${LoaderBounce} 0.6s -0.14s infinite ease-in-out;
    }

    &:nth-child(2) {
        animation: ${LoaderBounce} 0.6s -0.07s infinite ease-in-out;
    }

    &:last-child {
        animation: ${LoaderBounce} 0.6s 0s infinite ease-in-out;
    }
`;

const Loader = () => (
    <Box width="66px">
        <LoaderBall />
        <LoaderBall />
        <LoaderBall />
    </Box>
);

export default Loader;
