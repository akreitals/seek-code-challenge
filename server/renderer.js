import React from 'react';
import ReactDomServer from 'react-dom/server';
import { ServerStyleSheet } from 'styled-components';
import { Provider } from 'react-redux';
import { StaticRouter } from 'react-router-dom';
import { configureStore } from '../app/state/configureStore';
import ThemeProvider from '../app/components/ThemeProvider/ThemeProvider';

import RoutesContainer from '../app/containers/Routes/RoutesContainer';

const isProduction = process.env.NODE_ENV === 'production' || false;

const render = (html, styles) => {
    const assetsManifest =
        process.env.webpackAssets && JSON.parse(process.env.webpackAssets);

    return `
    <!DOCTYPE html>
    <html lang="en">
        <head>
            <meta charset="UTF-8">
            <meta name="viewport" content="width=device-width, initial-scale=1">
            <meta http-equiv="x-ua-compatible" content="ie=edge">
            <title>Seek Coding Challenge</title>
            ${styles}
        </head>
        <body>
            <div id="root">${html}</div>
            <script src='${
                isProduction ? assetsManifest['/vendor.js'] : 'js/vendor.js'
            }'></script>
            <script src='${
                isProduction ? assetsManifest['/app.js'] : 'js/app.js'
            }'></script>
        </body>
    </html>
    `;
};

const serverSideRender = (req, res) => {
    const context = {};
    const sheet = new ServerStyleSheet();
    const store = configureStore();

    const appWithRouter = (
        <Provider store={store}>
            <ThemeProvider>
                <StaticRouter location={req.url} context={context}>
                    <RoutesContainer />
                </StaticRouter>
            </ThemeProvider>
        </Provider>
    );

    if (context.url) {
        return res.redirect(context.url);
    }

    const html = ReactDomServer.renderToString(
        sheet.collectStyles(appWithRouter)
    );

    const styles = sheet.getStyleTags();

    return res.status(200).send(render(html, styles));
};

export default serverSideRender;
