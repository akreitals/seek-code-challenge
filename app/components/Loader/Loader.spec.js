import React from 'react';
import renderer from 'react-test-renderer';
import Loader from './Loader';

describe('Loader', () => {
    test('should render a <Loader /> component', () => {
        const json = renderer.create(<Loader />).toJSON();
        expect(json).toMatchSnapshot();
    });
});
