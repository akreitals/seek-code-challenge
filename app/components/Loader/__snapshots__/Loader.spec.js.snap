// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader should render a <Loader /> component 1`] = `
<div
  className="sc-bdVaJa kBjoPC"
  width="66px"
>
  <div
    className="sc-13b099f-0 iQGCCd"
  />
  <div
    className="sc-13b099f-0 iQGCCd"
  />
  <div
    className="sc-13b099f-0 iQGCCd"
  />
</div>
`;
